cmd/ig/daemon.go

```diff
@@ -100,11 +127,57 @@ func newDaemonCommand(runtime runtime.Runtime) *cobra.Command {
 			log.Warnf("reading config: %v", err)
 		}
 
+		tlsOptionsSet := 0
+		for _, tlsOption := range []string{serverKey, serverCert, clientCA} {
+			if len(tlsOption) != 0 {
+				tlsOptionsSet++
+			}
+		}
+
+		if tlsOptionsSet > 1 && tlsOptionsSet < 3 {
+			return fmt.Errorf(`
+missing at least one the TLS related options:
+	* tls-key: %q
+	* tls-cert: %q
+	* tls-client-ca: %q
+All these options should be set at the same time to enable TLS connection`,
+				serverKey, serverCert, clientCA)
+		}
```

Comment:
It's fine now. Later on we can improve it to allow only client or server verification if we want to